api/fs.go

```diff
@@ -0,0 +1,125 @@
+package api
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/hashicorp/nomad/client/allocdir"
+	"io"
+	"net/http"
+	"net/url"
+	"strconv"
+)
+
+// AllocFS is used to introspect an allocation directory on a Nomad client
+type AllocFS struct {
+	client *Client
+}
+
+// AllocFS returns an handle to the AllocFS endpoints
+func (c *Client) AllocFS() *AllocFS {
+	return &AllocFS{client: c}
+}
+
+// List is used to list the files at a given path of an allocation directory
+func (a *AllocFS) List(alloc *Allocation, path string, q *QueryOptions) ([]*allocdir.AllocFileInfo, *QueryMeta, error) {
```

Comment:
@dadgar The thing is we are doing the prefix stuff with the alloc id in the cli, so we have the allocation object beforehand.